src/main/scala/com/codacy/parsers/CoverageParser.scala

```diff
@@ -15,21 +15,37 @@ trait CoverageParser {
 
 object CoverageParser {
 
-  val parsers: List[CoverageParser] =
+  val allParsers: List[CoverageParser] =
     List(CoberturaParser, JacocoParser, CloverParser, OpenCoverParser, DotcoverParser, PhpUnitXmlParser, LCOVParser)
 
-  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] = {
+  def parse(
+      projectRoot: File,
+      reportFile: File,
+      forceParser: Option[CoverageParser] = None
+  ): Either[String, CoverageReport] = {
     val isEmptyReport = {
       // Just starting by detecting the simplest case: a single report file
       Try(reportFile.isFile && reportFile.length() == 0).getOrElse(false)
     }
 
+    val parsers = forceParser match {
+      case Some(parser) => List(parser)
+      case _ => allParsers
+    }
+
+    object ParsedCoverage {
+      def unapply(parser: CoverageParser): Option[CoverageReport] = {
+        parser.parse(projectRoot, reportFile).toOption
+      }
+    }
+
     if (isEmptyReport) {
       Left(s"Report file ${reportFile.getCanonicalPath} is empty")
     } else {
-      parsers.toStream
-        .map(_.parse(projectRoot, reportFile))
-        .find(_.isRight)
+      parsers.view
+        .collectFirst {
+          case ParsedCoverage(report: CoverageReport) => Right(report)
```

Comment:
It would be cool if we logged the "chosen" parser. This way we can know just from the logs and avoid asking users to use the force mechanism

Reply:
I like this suggestion, let me try to implement it without breaking the existing signature ...

Reply:
done